### PR TITLE
Add initial step status handling

### PR DESCRIPTION
**Pull request by rick-a-lane-ii**
Add initial step status handling

### CODE REVIEW

File: pkg/metadataapi/server/api/status.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	utilapi "github.com/puppetlabs/leg/httputil/api"
+	"github.com/puppetlabs/relay-core/pkg/metadataapi/errors"
+	"github.com/puppetlabs/relay-core/pkg/metadataapi/server/middleware"
+	"github.com/puppetlabs/relay-core/pkg/model"
+)
+
+type GetActionStatusResponseEnvelope struct {
+	ExitCode int    `json:"exit_code"`
+	Status   string `json:"status"`
```
Comment (impl) [inline pkg/metadataapi/server/api/status.go:15]: Think this is unused?

File: pkg/manager/memory/actionstatus.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func (s *ActionStatusManager) Set(ctx context.Context, sm *model.ActionStatus) error {
+	s.m.Set(ActionStatusKey{ActionHash: s.me.Hash().String()}, sm)
+	return nil
+}
+
+var _ model.ActionStatusManager = &ActionStatusManager{}
+
+func NewActionStatusManager(step *model.Step, backend *ActionStatusMap) *ActionStatusManager {
```
Comment (impl) [inline pkg/manager/memory/actionstatus.go:56]: I think this could take any action, not just a step, right?

**Review by impl — APPROVED** [review]
LGTM

File: pkg/metadataapi/server/api/status.go (hunk 1 above)
Comment (rick-a-lane-ii) [reply pkg/metadataapi/server/api/status.go:15]: Ahh, nice catch. I stripped out the "status" fields for now (but missed this one), as I want to be very deliberate about how/where the status gets set, and I haven't worked up that design yet.

File: pkg/manager/memory/actionstatus.go (hunk 2 above)
Comment (rick-a-lane-ii) [reply pkg/manager/memory/actionstatus.go:56]: Yeah, it originally only supported steps to have parity with the ConfigMap implementation. But I removed it at the last moment; I keep going back and forth on whether it should be similarly restricted or not. Considering we're making the general design to be viable for all actions.

Comment (rick-a-lane-ii) [reply pkg/manager/memory/actionstatus.go:56]: Ahh, and I meant the in-memory `Set` which is currently not restricting the Action type, as opposed to the ConfigMap implementation. I didn't even notice that the `step *model.Step` still needs to be updated, thanks.

**Pull request merged**